backend/nlp.py

```diff
@@ -141,21 +178,27 @@ def generate_timeline_json(sections, title, quiet, unpruned, percentile):
 
     unnormalised_matrices = []
     character_lists = []
+    first_interactions_overall = {}
+    first_interactions_per_char = {}
     for (i, section) in enumerate(sections):
         if not quiet:
             print("Analysing section {} of {}...".format(i + 1, len(sections)))
         interactions, characters = generate_interactions_matrix(
-            section, interactions, characters, {})
+            section, interactions, characters, {}, first_interactions_overall, first_interactions_per_char)
```

Comment:
We seem to make a new `characters_dict` each interaction instead of building upon the previous one